### PR TITLE
feat(performance-issues): add ability to query transactions by group_ids

### PR DESCRIPTION
**Pull request by udameli**
This PR adds an ability to query transaction events by a single group id, so `events-stats`, `events` and `event-span-count-histogram` endpoints can query affected transactions in the Issue Details page. This PR is dependent on https://github.com/getsentry/snuba/pull/3066 and https://github.com/getsentry/snuba/pull/3104 to be merged first so that group ids can be saved on transactions.

### CODE REVIEW

File: src/sentry/snuba/events.py [hunk 1]
```diff
@@ -26,6 +26,16 @@ class Columns(Enum):
         discover_name="group_id",
         alias="issue.id",
     )
+    # This is needed to query transactions by group id
+    # in the Issue Details page. This will not be
+    # exposed to users through discover search.
```
Comment (wedamija) [inline src/sentry/snuba/events.py:31]: I think we might need to allow discover to filter on this, to support both `open in discover`, and when we switch the `events` tab to use discover.

I think that in general, it's ok to allow us to filter on `group_ids` in discover, but not select it as a column and group by, etc.

Comment (udameli) [reply src/sentry/snuba/events.py:31]: @wmak is there a reason why we can't expose `group_ids` in discover? want to make sure I don't break anything for discover here 🙏

Comment (udameli) [reply src/sentry/snuba/events.py:31]: based on the offline chat with Will, `group_ids` hasn't been added to the `discover` dataset but as long we force Discover search to look into the transactions dataset by setting `event.type: transaction` we should be able to search by `group ids`

Comment (udameli) [reply src/sentry/snuba/events.py:31]: and that should work for `open in discover` and `events` tab 👍

File: tests/snuba/api/endpoints/test_organization_events_stats.py [hunk 2]
```diff
@@ -563,6 +563,23 @@ def test_project_id_query_filter(self):
...
+        # store transaction with group ids
+        _data = load_data("transaction")
+        # _data["group_ids"] = [123]
+        self.store_event(_data, self.project2.id)
+
+        response = self.do_request(
+            data={
+                "end": iso_format(before_now()),
+                "start": iso_format(before_now(hours=2)),
+                "query": "event.type:transaction performance.issue_ids:123",
```
Comment (udameli) [inline tests/snuba/api/endpoints/test_organization_events_stats.py:576]: even when I specify `event.type:transaction` the query still searches in the `discover` dataset 🤔 
```
 {'app_id': 'default',
 'dataset': 'discover',
 'query': 'MATCH (discover) SELECT count() AS `count` BY time WHERE type = '
          "'transaction' AND ifNull(tags[performance.issue_ids], '') = '123' "
          "AND timestamp >= toDateTime('2022-08-30T18:56:22') AND timestamp < "
          "toDateTime('2022-08-30T20:56:22') AND project_id IN array(3, 2) "
          'ORDER BY time ASC LIMIT 10000 GRANULARITY 1800'}
api.organization-event-stats.sql:
 SELECT (toDateTime(multiply(intDiv(toUInt32(finish_ts), 1800), 1800), 'Universal') AS _snuba_time), (count() AS _snuba_count)
  FROM transactions_local
 WHERE equals(('transaction' AS _snuba_type), 'transaction')
   AND has(_tags_hash_map, cityHash64('performance.issue_ids=123'))
   AND greaterOrEquals((finish_ts AS _snuba_finish_ts), toDateTime('2022-08-30T18:56:22', 'Universal'))
   AND less(_snuba_finish_ts, toDateTime('2022-08-30T20:56:22', 'Universal'))
   AND in((project_id AS _snuba_project_id), [3, 2])
 GROUP BY _snuba_time
 ORDER BY _snuba_time ASC
 LIMIT 10000
OFFSET 0
```

@wmak What am I missing here?

Comment (wmak) [reply tests/snuba/api/endpoints/test_organization_events_stats.py:576]: I believe you have to add group_ids [here in snuba](https://github.com/getsentry/snuba/blob/master/snuba/datasets/entities/discover.py#L247)

And I might have been wrong about [only needing group_ids on transactions, you may need it in discover too](https://github.com/getsentry/sentry/pull/38024/files#diff-1f77bf31e7ad04aa2f505c69d362ce4f087c18e439c3a769829506a2196472e5R36)

Comment (udameli) [reply tests/snuba/api/endpoints/test_organization_events_stats.py:576]: ok, snuba PR here https://github.com/getsentry/snuba/pull/3104

File: src/sentry/search/events/builder.py [hunk 3]
```diff
@@ -1109,13 +1109,13 @@ def _default_filter_converter(self, search_filter: SearchFilter) -> Optional[Whe
                     .replace("_", "\\_")
                     .replace("*", "%"),
                 )
-            elif name in ARRAY_FIELDS and search_filter.is_in_filter:
+            elif search_filter.is_in_filter:
```
Comment (udameli) [inline src/sentry/search/events/builder.py:1112]: we're already checking for `if name in ARRAY_FIELDS` above

File: tests/snuba/api/endpoints/test_organization_events.py [hunk 4]
```diff
@@ -493,6 +495,132 @@ def test_auto_insert_project_name_when_event_id_present_with_aggregate(self):
...
+            self.store_event(data=data, project_id=self.project.id)
+
+        query = {
+            "field": ["count()"],
+            "statsPeriod": "1h",
+            "query": f"performance.issue_ids:{_perf_group.id}",
+        }
+        response = self.do_request(query)
+        assert response.status_code == 200, response.content
+        assert response.data["data"][0]["count()"] == 1
```
Comment (udameli) [inline tests/snuba/api/endpoints/test_organization_events.py:522]: I don't know why but this test passes when I run just this function but fails when I run the entire suite

Comment (udameli) [reply tests/snuba/api/endpoints/test_organization_events.py:527]: this was due to `int(v, 16)` that was messing with group ids. this is resolved now

**Review by wedamija — APPROVED** [review]

**Pull request merged**